pulser/parametrized/paramobj.py

```diff
@@ -99,12 +100,30 @@ def build(self):
         return self._instance
 
     def _to_dict(self):
+        def class_to_dict(cls):
+            return obj_to_dict(self, _build=False, _name=cls.__name__,
+                               _module=cls.__module__)
+
+        args = list(self.args)
         if isinstance(self.cls, Parametrized):
             cls_dict = self.cls._to_dict()
+        elif (hasattr(args[0], self.cls.__name__)
+              and inspect.isfunction(self.cls)):
+            # Check for parametrized methods
+            if inspect.isclass(self.args[0]):
+                # staticmethod or classmethod
```

Comment:
In the case of static methods, are you sure the first arg is also a cls ?

Reply:
You're right, I was thinking. This will only work for `classmethods`, I've fixed this.